nf_core/components/create.py

```diff
@@ -411,3 +430,160 @@ def _get_username(self):
+            )
+        # Create a nextflow.config file if it contains information other than publishDir
+        pytest_dir = Path(self.directory, "tests", self.component_type, self.org, self.component_dir)
+        nextflow_config = pytest_dir / "nextflow.config"
+        if nextflow_config.is_file():
+            with open(nextflow_config, "r") as fh:
+                config_lines = ""
+                for line in fh:
+                    if "publishDir" not in line:
+                        config_lines += line
+            if len(config_lines) > 0:
+                log.debug("Copying nextflow.config file from pytest tests")
+                with open(
+                    Path(self.directory, self.component_type, self.org, self.component_dir, "tests", "nextflow.config"),
+                    "w+",
+                ) as ofh:
+                    ofh.write(config_lines)
+
+    def _collect_pytest_tests(self):
+        pytest_dir = Path(self.directory, "tests", self.component_type, self.org, self.component_dir)
+        tests = []
+        name = None
+        input = None
+        in_input = False
+        input_number = 0
+        number_of_inputs = []
+        with open(pytest_dir / "main.nf") as fh:
+            for line in fh:
+                if line.strip().startswith("workflow"):
+                    # One test
+                    if name and input:
+                        tests.append((name, input))
+                        name = None
+                        input = None
+                    name = line.split()[1]
+                elif line.strip().startswith("input"):
+                    # First input
+                    input = [line.split("=")[1]]
+                    in_input = True
+                    number_of_inputs.append(1)
+                elif "=" in line and "nextflow.enable.dsl" not in line:
+                    # We need another input
+                    input_number += 1
+                    in_input = True
+                    input.append(line.split("=")[1])
+                    number_of_inputs[input_number] += 1
+                elif in_input:
+                    # Retrieve all lines of an input
+                    if self.component_dir.replace("/", "_").upper() in line:
+                        in_input = False
+                        continue
+                    input[input_number] += line
+
+        if name and input:
+            tests.append((name, input))
+
+        if max(number_of_inputs) > 1:
+            # Check that all tests have the same number of inputs
+            for test in tests:
+                if len(test[1]) < max(number_of_inputs):
+                    for i in range(max(number_of_inputs) - len(test[1])):
+                        test[1].append(" []")
+
+        return tests
+
+    def _create_new_test(self, name, inputs):
+        input_string = ""
+        for i, input in enumerate(inputs):
```

Comment:
any chance we can use the template for this so we don't need to keep track of things changing in two places?